Sources/SourceControl/GitRepository.swift

```diff
@@ -209,8 +209,8 @@ public class GitRepository: Repository, WorkingCheckout {
 
     // MARK: Repository Interface
 
-    public var tags: [String] { return tagsCache.getValue(self) }
-    private var tagsCache = LazyCache(getTags)
+    // FIXME: Tags can be cached as long as the cache is cleared after the fetch() call.
+    public var tags: [String] { return getTags() }
```

Comment:
We will definitely want to get this cache back...

Now that we have Dispatch available, I would probably put a serial queue in each repo object and use it to order all the operations. Ideally we would go ahead and fix this and add fetch() as a precursor to this patch.